python/lsst/daf/butler/queries/expression_factory.py

```diff
@@ -233,6 +254,42 @@ def _expression(self) -> tree.ColumnExpression:
         return self._expr
 
 
+class BooleanScalarExpressionProxy(ScalarExpressionProxy):
+    """A `ScalarExpressionProxy` representing a boolean column.  You should
+    call `as_boolean()` on this object to convert it to an instance of
+    `Predicate` before attempting to use it.
+
+    Parameters
+    ----------
+    expression : `.tree.ColumnReference`
+        Boolean column reference that backs this proxy.
+    """
+
+    # This is a hack/work-around to make static typing work when referencing
+    # dimension record metadata boolean columns.  From the perspective of
+    # typing, anything boolean should be a `Predicate`, but the type system has
+    # no way of knowing whether a given column is a bool or some other type.
+
+    def __init__(self, expression: tree.ColumnReference) -> None:
+        if expression.column_type != "bool":
+            raise ValueError(f"Expression is a {expression.column_type}, not a 'bool': {expression}")
+        self._boolean_expression = expression
+
+    @property
+    def is_null(self) -> tree.Predicate:
+        return ResolvedScalarExpressionProxy(self._boolean_expression).is_null
+
+    def as_boolean(self) -> tree.Predicate:
```

Comment:
I don't like forcing users to call `as_boolean()` to get a Predicate out of this, but I didn't really have a better idea. 

We can't just return a Predicate directly from `DimensionElementProxy.__getattr__` because it would completely destroy the static typing.  The return type would become `ScalarExpressionProxy | Predicate` and those types have almost no methods in common.